src/app/resources/resources-view.component.ts

```diff
@@ -41,27 +43,48 @@ export class ResourcesViewComponent implements OnInit {
   urlPrefix = environment.couchAddress + 'resources/';
 
   ngOnInit() {
-    this.route.paramMap.pipe(switchMap((params: ParamMap) => this.getResource(params.get('id'))))
-      .subscribe(resource => this.resource = resource);
+    this.route.paramMap.pipe(switchMap((params: ParamMap) => this.getResource(params.get('id'), params.get('nationname'))))
+      .subscribe(resource => this.resource = resource );
   }
 
-  getResource(id: string) {
-    return this.couchService.get('resources/' + id)
-      .then((data) => {
-        // openWhichFile is used to label which file to start with for HTML resources
-        const filename = data.openWhichFile || Object.keys(data._attachments)[0];
-        this.mediaType = data.mediaType;
-        this.contentType = data._attachments[filename].content_type;
-        this.resourceSrc = this.urlPrefix + data._id + '/' + filename;
-        if (this.mediaType === 'pdf') {
-          this.pdfSrc = this.sanitizer.bypassSecurityTrustResourceUrl(this.resourceSrc);
-        }
-        if (this.mediaType === 'zip') {
-          this.router.navigate([ '/resources' ]);
-          window.open(this.resourceSrc);
-        }
-        return data;
-      }, (error) => console.log('Error'));
+  getResource(id: string, nationname: string) {
+    if (nationname !== null) {
+      return this.couchService.get('nations/_all_docs?include_docs=true')

```

Comment:
Code can be simplify and also I tested with the unknown nation and it fetching the data.